src/OMSimulatorLib/Types.h

```diff
@@ -86,6 +102,12 @@ typedef enum {
   oms_component_port      ///< port
 } oms_element_type_enu_t;
 
+typedef enum {
+  oms_system_tlm,      ///< TLM System
+  oms_system_wc,       ///< Weakly Coupled System
+  oms_system_sc        ///< Strongly Coupled System
+} oms_system_enu_t;
```

Comment:
This could be removed in favour of the new type `oms3_element_type_enu_t`.

Reply:
Or we keep it and change `oms3_element_type_enu_t` to:
```c
typedef enum {
  oms_element_none,
  oms_element_system,
  oms_element_component
} oms3_element_type_enu_t;
```

Reply:
This is much better as it doesn't break existing `oms3_addSystem` API.
But `oms_element_component` means you want to introduce a `oms_component_enu_t` as well, right?

Reply:
Exactly, see #348.